redaxo/src/core/tests/sql/sql_test.php

```diff
@@ -183,6 +183,25 @@ public function testSetGetArrayValue()
         static::assertEquals([1, 2, 3], $sql->getArrayValue('col_array'), 'get a previous set array');
     }
 
+    public function testGetValueWithType()
+    {
+        $sql = rex_sql::factory();
+        $sql->setTable(self::TABLE);
+        $sql->setValue('col_str', 'abc');
+        $sql->setValue('col_int', '5');
+        $sql->setArrayValue('col_array_1', [3, '2', null]);
+        $sql->setArrayValue('col_array_2', ['foo' => 3, 'bar' => 2]);
+
+        static::assertSame('abc', $sql->getValue('col_str', 'string'));
+        static::assertSame(5, $sql->getValue('col_int', 'int'));
+        static::assertSame([3, '2', null], $sql->getValue('col_array_1', 'array'));
+        static::assertSame([3, 2, 0], $sql->getValue('col_array_1', 'array[int]'));
```

Comment:
Hmm ich seh grad dass wir getValue mit array eigentlich gar nicht unterstützen.. sondern nur getArrayValue.. von der db kann man ja kein natives array bekommen

Reply:
ja, bisher konnte da kein array rauskommen, aber ich dachte, es wäre praktisch, wenn man hier auch die array-types verwenden könnte. deswegen ja dieses json_decode.
Führt das zu Problemen?

Reply:
Ich finde es inkonsistent bzgl getArrayValue().

Aufgefallen ist es mir in https://github.com/redaxo/redaxo/pull/4150 da es dann keinen sinn mehr macht den type von getValue auf `scalar` zu ändern.

Ein array kann man ja nur dann drinnen haben wenn man es mit setValue() reinsetzt (aber speichern kann man dann dem rex_sql auch nicht mehr, weil er im value-building nicht mit arrays umgehen kann

Reply:
Also man müsste dann den wert mit setArrayValue() schreiben um ihn mit getValue(.., array) auszulesen.

Das ist unschön bzgl symmetrie